pkg/alertmanager/alertmanager.go

```diff
@@ -203,6 +199,13 @@ func New(cfg *Config, reg *prometheus.Registry) (*Alertmanager, error) {
 	c = am.state.AddState("sil:"+cfg.UserID, am.silences, am.registry)
 	am.silences.SetBroadcast(c.Broadcast)
 
+	// State replication needs to be started after the state keys are defined.
+	if service, ok := am.state.(services.Service); ok {
+		if err := service.StartAsync(context.Background()); err != nil {
```

Comment:
We don't wait until started here (and it's correct). However, this means that we may start using this `Alertmanager` instance before settlement is completed (and I believe this is not correct). Am I missing anything?

Reply:
I have been assuming we want to settle in the background, because an `Alertmanager` is spun up for every tenant. If they all have to hit the timeout, that might take too long if done in serial. That being said, it's harder to reason about correctness in this case

Perhaps safer to change it to wait for now, and explore doing it in the background as a separate piece of work?

Reply:
It's worth noting that "start using" is not as one would think. Yes, we'll accept alerts, silences, etc. However, we'll wait for the state to be replicated before we send a notification.

Reply:
Will leave as-is. My (current) understanding is as Josh said - there is no requirement to block (except for notifications, which are blocked via the call into `WaitReady`).